fastlane/Fastfile

```diff
@@ -84,6 +92,136 @@ def gutenberg_config!
+  code_freeze_build_code = VERSION_CALCULATOR.next_release_version(version: current_release_version)
+  # Return the formatted build code
+  BUILD_CODE_FORMATTER.build_code(version: code_freeze_build_code)
+end
+
+# Returns the internal build code of the app for the code freeze. It is the release version name plus sets the build
+# number to 0
+#
+def code_freeze_internal_build_code
+  # Read the current build code from the .xcconfig file and parse it into an AppVersion object
+  # The AppVersion is used because WP/JPiOS uses the four part (1.2.3.4) build code format, so the version
+  # calculator can be used to calculate the next four-part version
+  current_release_version = VERSION_FORMATTER.parse(PUBLIC_VERSION_FILE.read_release_version)
+  # Calculate the next release version, which will be used as the basis of the new build code
+  next_release_version = VERSION_CALCULATOR.next_release_version(version: current_release_version)
+  code_freeze_build_code = INTERNAL_BUILD_CODE_CALCULATOR.next_build_code(version: next_release_version)
+  # Return the formatted build code
+  BUILD_CODE_FORMATTER.build_code(version: code_freeze_build_code)
+end
+
+# Returns the build code of the app for the code freeze. It is the hotfix version name plus sets the build number to 0
+#
+def hotfix_build_code(release_version:)
+  version = VERSION_FORMATTER.parse(release_version)
+  # Return the formatted build code
+  BUILD_CODE_FORMATTER.build_code(version:)
+end
+
+# Returns the next build code of the app
+#
+def next_build_code
+  # Read the current build code from the .xcconfig file and parse it into an AppVersion object
+  # The AppVersion is used because WP/JPiOS uses the four part (1.2.3.4) build code format, so the version
+  # calculator can be used to calculate the next four-part version
+  current_build_code = VERSION_FORMATTER.parse(PUBLIC_VERSION_FILE.read_build_code(attribute_name: 'VERSION_LONG'))
+  # Calculate the next build code
+  next_build_code = VERSION_CALCULATOR.next_build_number(version: current_build_code)
+  # Return the formatted build code
+  BUILD_CODE_FORMATTER.build_code(version: next_build_code)
+end
+
+# Returns the next internal build code of the app
+#
+def next_internal_build_code
+  # Read the current build code from the .xcconfig file and parse it into an AppVersion object
+  # The AppVersion is used because WP/JPiOS uses the four part (1.2.3.4) build code format, so the version
+  # calculator can be used to calculate the next four-part version
+  current_build_code = VERSION_FORMATTER.parse(PUBLIC_VERSION_FILE.read_build_code(attribute_name: 'VERSION_LONG'))
+  # Calculate the next build code
+  next_build_code = INTERNAL_BUILD_CODE_CALCULATOR.next_build_code(version: current_build_code)
+  # Return the formatted build code
+  BUILD_CODE_FORMATTER.build_code(version: next_build_code)
+end
+
+# Returns the next internal hotfix build code of the app
+#
+def internal_hotfix_build_code(release_version:)
```

Comment:
The "current" methods use the "current [internal] name" structure. What do you think of updating this one for consistency?

```suggestion
def hotfix_internal_build_code(release_version:)
```

---

Speaking of consistency, I find it useful to optimize names for autocompletion by keeping the common parts a the root and suffixing with the differentiation. In practice:

def release_version_current
...

def release_version_current_internal
...

def release_version_next
...

def build_code_current
...

def build_code_current_internal
...

# etc
```

What do you think?

Reply:
I agree that changing those names makes the readability and consistency better. I changed the names in https://github.com/wordpress-mobile/WordPress-iOS/commit/7a4ea4a6ee333f25de966fabaeb84ab913afcf62